### ISSUE TITLE
Password with special characters is not saved correctly on application pool

### ISSUE THREAD

**Issue by alfhenrik** [OctopusDeploy/Issues#2169]
Quite a few customers reporting having issues with the password that is passed when setting up an application pool with a specific user.

Sources: 
http://help.octopusdeploy.com/discussions/problems/24111
http://help.octopusdeploy.com/discussions/problems/26003

**Comment by alfhenrik** [comment]
Customer having issues with Windows Service custom password (no special characters) that is bound to variables as well. Can we have a look at that while we're in that area :)

Source: `bcall`s reply in http://help.octopusdeploy.com/discussions/problems/29411

**Comment by alfhenrik** [comment]
Another report that bound variable in step template doesn't get updated when promoted through environments

```
We are running into this issue with a create scheduled task step (from the community library). We have a variable containing the password that is scoped to each environment, and the password configured for the step is bound to that variable (custom expression #{MyPasswordVariable}). After deploying to the first environment, each subsequent deployment to other environments uses the password from the first environment, but all the other scoped variables are changed correctly. We are already running 3.0.24. I have checked the release notes up to the current version and it doesn't appear that this issue has been fixed?
```

**Comment by robpearson** [comment]
Unfortunately, I failed to reproduce any of the scenarios reported in this issue.  I'm sharing the details to ensure there isn't a gap in my testing.

I set up a brand new Octopus server running 3.2.8 and added a single tentacle with the matching version.  I created the following test accounts (username: password) on the target server.  

* BugUser01: Password*
* BugUser02: Password#
* BugUser03: Password>
* BugUser04: Password$%^&

I created dev/test/staging and prod environments using the same tentacle but varied website/service names to ensure uniqueness. I created a template project and populated the variables with these details as per the following screenshot. Then I cloned the project for each of the test scenarios.  

![2015-12-08_13-33-24](https://cloud.githubusercontent.com/assets/216245/11703810/ea50ec6a-9f2d-11e5-9825-4bba0684d44b.png)

## Test Scenarios
* Project w/ Nuget package containing a simple website.  Binding app pool to custom user accounts per env.
  - Used bound non-sensitive variables.
  - Used bound sensitive variables.
* Project w/ Nuget package containing a simple windows service.  Binding app pool to custom user accounts per env.
  - Used bound non-sensitive variables.
  - Used bound sensitive variables.
* Project w/ [App Pool Template](https://library.octopusdeploy.com/#!/step-template/actiontemplate-iis-apppool-create).  
  - Used bound prompted variables.
  - Used bound sensitive variables.

**Comment by robpearson** [comment]
NOTE: My first windows service deployment failed with the following error.

`Start-Service : Service 'dev Service (devService)' cannot be started due to the following error: Cannot start service devService on computer '.'`

I checked the service on the target service and it existed but it was stopped.  I tried to restart it and it displayed a logon failed error.  I manually copy-pasted the password and clicked ok and it displayed a message saying the user account had been granted the `Log on as a service` right.  I then updated the remaining accounts the deployments succeeded.
